lib/tpm2.c

```diff
@@ -1309,3 +1349,164 @@ tool_rc tpm2_unseal(ESYS_CONTEXT *esys_context, tpm2_loaded_object *sealkey_obj,
+
+    TSS2_RC rval = Esys_SelfTest(ectx, ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+        fullTest);
+    if (rval != TPM2_RC_SUCCESS) {
+        LOG_PERR(Esys_SelfTest, rval);
+        return tool_rc_from_tpm(rval);
+    }
+
+    return tool_rc_success;
+}
+
+tool_rc tpm2_gettestresult(ESYS_CONTEXT *ectx, TPM2B_MAX_BUFFER **outData,
+        TPM2_RC *testResult) {
+
+    TSS2_RC rval = Esys_GetTestResult(ectx, ESYS_TR_NONE, ESYS_TR_NONE,
+            ESYS_TR_NONE, outData, testResult);
+    if (rval != TSS2_RC_SUCCESS) {
+        LOG_PERR(Esys_GetTestResult, rval);
+        return tool_rc_from_tpm(rval);
+    }
+
+    return tool_rc_success;
+}
+
+tool_rc tpm2_loadexternal(ESYS_CONTEXT *ectx,
+        const TPM2B_SENSITIVE *inPrivate,
+        const TPM2B_PUBLIC *inPublic,
+        TPMI_RH_HIERARCHY hierarchy,
+        ESYS_TR *objectHandle) {
+
+    TSS2_RC rval = Esys_LoadExternal(ectx,
+            ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+            inPrivate, inPublic, hierarchy,
+            objectHandle);
+    if (rval != TSS2_RC_SUCCESS) {
+        LOG_PERR(Esys_LoadExternal, rval);
+        return tool_rc_from_tpm(rval);
+    }
+
+    return tool_rc_success;
+}
+
+tool_rc tpm2_pcr_event(ESYS_CONTEXT *ectx,
+        ESYS_TR pcr, tpm2_session *session,
+        const TPM2B_EVENT *eventData,
+        TPML_DIGEST_VALUES **digests) {
+
+    ESYS_TR shandle1 = ESYS_TR_NONE;
+    tool_rc rc = tpm2_auth_util_get_shandle(ectx, pcr, session,
+            &shandle1);
+    if (rc != tool_rc_success) {
+        return rc;
+    }
+
+    TSS2_RC rval = Esys_PCR_Event(ectx, pcr, shandle1, ESYS_TR_NONE,
+            ESYS_TR_NONE, eventData, digests);
+    if (rval != TSS2_RC_SUCCESS) {
+        LOG_PERR(Esys_PCR_Event, rval);
+        return tool_rc_from_tpm(rval);
+    }
+
+    return tool_rc_success;
+}
+
+tool_rc tpm2_getrandom(ESYS_CONTEXT *ectx, UINT16 bytesRequested,
+        TPM2B_DIGEST **randomBytes) {
+
+    TSS2_RC rval = Esys_GetRandom(ectx, ESYS_TR_NONE, ESYS_TR_NONE,
+        ESYS_TR_NONE, bytesRequested, randomBytes);
+    if (rval != TPM2_RC_SUCCESS) {
+        LOG_PERR(Esys_GetRandom, rval);
+        return tool_rc_from_tpm(rval);
+    }
+
+    return tool_rc_success;
+}
+
+tool_rc tpm2_startup(ESYS_CONTEXT *ectx, TPM2_SU startupType) {
+
+    TSS2_RC rval = Esys_Startup(ectx, startupType);
+    if (rval != TPM2_RC_SUCCESS && rval != TPM2_RC_INITIALIZE) {
+        LOG_PERR(Esys_Startup, rval);
+        return tool_rc_from_tpm(rval);
+    }
+
+    return tool_rc_success;
+}
+
+tool_rc tpm2_pcr_reset(ESYS_CONTEXT *ectx,
+        ESYS_TR pcrHandle) {
+
+    TSS2_RC rval = Esys_PCR_Reset(ectx, pcrHandle, ESYS_TR_PASSWORD,
+            ESYS_TR_NONE, ESYS_TR_NONE);
+    if (rval != TSS2_RC_SUCCESS) {
+        LOG_PERR(Esys_PCR_Reset, rval);
+        return tool_rc_from_tpm(rval);
+    }
+
+    return tool_rc_success;
+}
+
+tool_rc tpm2_makecredential(ESYS_CONTEXT *ectx, ESYS_TR handle,
```

Comment:
every function argument needn't be on the newline, collapse this to character width per coding guideline.